touchforms/formplayer/experiments.py

```diff
@@ -0,0 +1,120 @@
+import laboratory
+import json
+import csv
+import datetime
+import logging
+from django.conf import settings
+import os
+
+diff_csv_filename = os.path.join(settings.FORMPLAYER_EXPERIMENT_DIRECTORY, 'formplayer_diff.csv')
+timing_csv_filename = os.path.join(settings.FORMPLAYER_EXPERIMENT_DIRECTORY, 'formplayer_timing.csv')
+
+
+class FormplayerExperiment(laboratory.Experiment):
+    logging = logging.getLogger(__name__)
+    session_id_mapping = {}
+
+    def publish(self, result):
+
+        if not os.path.exists(os.path.dirname(settings.FORMPLAYER_EXPERIMENT_DIRECTORY)):
+            os.makedirs(os.path.dirname(settings.FORMPLAYER_EXPERIMENT_DIRECTORY))
+
+        # if we're starting a new form, we need to store the mapping between session_ids so we can use later
+        if (self.name == "new-form"):
+            control_session_id = json.loads(result.control.value)["session_id"]
+            candidate_session_id = json.loads(result.observations[0].value)["session_id"]
+            FormplayerExperiment.session_id_mapping[control_session_id] = candidate_session_id
+
+        control = result.control
+        # We're only ever returning one of these (I think)
+        candidate = result.observations[0]
+
+        emit_timing_csv(control, candidate, self.name)
+
+        control_value = json.loads(result.control.value)
+        candidate_value = json.loads(result.observations[0].value)
+
+        if not formplayer_compare(control_value, candidate_value):
+            emit_diff_csv(control_value, candidate_value, self.context)
+
+
+def emit_diff_csv(control_value, candidate_value, context):
```

Comment:
this is far more performant as a logger. we can setup the format for the logger to log all the attributes as comma separated. i'm worried opening and closing a file on every request (which is a lot for touchforms) will be pretty brutal. using a logger will also simplify this logic